src/snowflake/snowpark/modin/plugin/_internal/pivot_utils.py

```diff
@@ -194,6 +194,20 @@ def pivot_helper(
     if ordered_dataframe.queries.get("post_actions"):
         ordered_dataframe = cache_result(ordered_dataframe)
 
+    if pivot_aggr_groupings is None:
+        # When pivot_aggr_groupings is None, there are no `values` to compute on. In that case, we simply return
+        # a DataFrame with no columns, whose index is the result of grouping by the index columns.
+        ordered_dataframe = ordered_dataframe.group_by(
+            groupby_snowflake_quoted_identifiers
+        )
+        return InternalFrame.create(
+            ordered_dataframe=ordered_dataframe,
+            data_column_pandas_index_names=[None] + columns,
```

Comment:
why there is an extra [None] here?

Reply:
I'm not actually certain myself - it seems that when there are no values columns, pandas adds an additional level to the index for the columns - I suspect its adding the level for the aggregation function?

Reply:
It seems that when there are no values, all rest of the columns that are not index + columns are used as values columns, and the index becomes multiindex, and all levels of the original columns are retained, which means the new data_column_pandas_index_names should be the original data_column_pandas_index_names + columns. you can try it with following example 
```
df = pd.DataFrame({'foo': ['one', 'one', 'one', 'two', 'two',
                           'three'],
                   'bar': ['A', 'B', 'C', 'A', 'B', 'C'],
                   'baz': [1, 2, 3, 4, 5, 6],
                   'zoo': [1, 2, 3, 1, 3, 5]})
df.columns.names = ['column']
df.pivot(index = 'bar', columns='foo')
```

This confirms with the original column index that is single level, you probably want to check if it the original column index is multi-level, and see if it follows the same reason.

Reply:
Sorry - to clarify, this is handling the case where there are no columns that can be used for values - e.g. in the above example, this codepath would only be hit if the call to `pivot_table` looked like this:
```python
df.pivot_table(index=['bar', 'baz'], columns=['foo', 'zoo'])
```

Reply:
I see, but i think for this part pivot and pivot_table behaves the same, you can use df.pivot_table(index = 'bar', columns='foo') with the above example for verify. i believe that is just a label expansion which retains the original name with expanded level

Reply:
i think it is still the same, i you try above example with the pivot_table operation, then do df2.columns, you see the following result
```
MultiIndex([], names=['column', 'foo', 'zoo'])
```

Reply:
Ah I see - seems like the name that is added should then be the current names of the index. Will add.

Reply:
Also confirmed that this is the case with MultiIndex:
```python
In [6]: df.columns = pd.MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1), ('b', 2)])

In [7]: df
Out[7]:
       a     b
       1  2  1  2
0    one  A  1  1
1    one  B  2  2
2    one  C  3  3
3    two  A  4  1
4    two  B  5  3
5  three  C  6  5

In [8]: df.columns.names = ['c1', 'c2']

In [9]: df
Out[9]:
c1      a     b
c2      1  2  1  2
0     one  A  1  1
1     one  B  2  2
2     one  C  3  3
3     two  A  4  1
4     two  B  5  3
5   three  C  6  5

In [10]: df.columns
Out[10]:
MultiIndex([('a', 1),
            ('a', 2),
            ('b', 1),
            ('b', 2)],
           names=['c1', 'c2'])

In [11]: df.pivot_table(index=[('a', 1), ('a', 2)], columns=[('b', 1), ('b', 2)]).columns
Out[11]: MultiIndex([], names=['c1', 'c2', ('b', 1), ('b', 2)])
```

Reply:
While adding the logic + tests for this, I decided to add tests to check that pivot_table works correctly with named columns when not empty (basically added a `named_columns` parameter to some other tests), and noticed that this causes a some other tests to fail (results from running all tests under the pivot directory):

`14 failed, 243 passed, 7 skipped, 26 xfailed, 4 xpassed, 164 warnings`

I'm thinking it makes sense to add the tests here (with an xfail) and fix the other failures not related to empty dataframes in a future PR. Thoughts?

Reply:
Also, side note: I've added tests for empty pivot table with multi-indexes to check that the multi index names are carried over to the result; however, we don't support pivot table with multi-index currently, so I've skipped the tests. I mostly just added them so that if we do add multi index support, we won't forget to turn on these tests and verify that they pass, but I'm happy to remove them for now? Do we prefer skips (which don't significantly increase CI time) or xfails with raises=NotImplementedError (which will make it so that these tests catch when we add multiindex support, but also add time to CI to run a bunch of tests we know will fail)?